experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/node.spec.ts

```diff
@@ -4,89 +4,149 @@
+			const child = root.object;
+			const p = node.parent(child);
+			assert.equal(node.parent(root.object), root);
+		});
 
-				const unsubscribe = api.on("subtreeChanging", (...args: any[]) => {
-					log.push(args);
-				});
+		itWithRoot("list", treeSchema, initialTree, (root) => {
+			assert.equal(node.parent(root.list), root);
+		});
 
-				mutate(root);
+		itWithRoot("root", treeSchema, initialTree, (root) => {
+			assert.equal(node.parent(root), undefined);
+		});
+	});
 
-				const numChanges = log.length;
-				assert(
-					numChanges > 0,
-					"Must receive change notifications after subscribing to event.",
-				);
+	describe("treeStatus", () => {
+		itWithRoot("object", treeSchema, initialTree, (root) => {
+			const o = root.object;
+			assert(o !== undefined);
+			assert.equal(node.status(o), TreeStatus.InDocument);
+			root.object = object.create({ content: 43 });
+			assert.equal(node.status(o), TreeStatus.Removed);
+		});
 
-				unsubscribe();
+		// TODO: Enable a test like this when lists are directly settable
+		// itWithRoot("list", treeSchema, initialTree, (root) => {
+		// 	const l = root.list;
+		// 	assert(l !== undefined);
+		// 	assert.equal(node.status(l), TreeStatus.InDocument);
+		// 	root.list = [];
+		// 	assert.equal(node.status(l), TreeStatus.Removed);
+		// });
+	});
 
-				mutate(root);
+	describe("events", () => {
+		function check(mutate: (root: ProxyRoot<typeof treeSchema>) => void) {
+			itWithRoot(
+				".on(..) must subscribe to change event",
+				treeSchema,
+				initialTree,
+				(root) => {
+					const log: any[][] = [];
+
+					node.on(root as SharedTreeNode, "afterChange", (...args: any[]) => {
+						log.push(args);
+					});
+
+					mutate(root);
+
+					const numChanges = log.length;
+					assert(
+						numChanges > 0,
+						"Must receive change notifications after subscribing to event.",
+					);
+				},
+			);
 
-				assert.equal(
-					log.length,
-					numChanges,
-					"Mutation after unsubscribe must not emit change events.",
-				);
-			});
+			itWithRoot(
+				".on(..) must return unsubscribe function",
+				treeSchema,
+				initialTree,
+				(root) => {
+					const log: any[][] = [];
+
+					const unsubscribe = node.on(
+						root as SharedTreeNode,
+						"afterChange",
+						(...args: any[]) => {
+							log.push(args);
+						},
+					);
+
+					mutate(root);
+
+					const numChanges = log.length;
+					assert(
+						numChanges > 0,
+						"Must receive change notifications after subscribing to event.",
+					);
+
+					unsubscribe();
+
+					mutate(root);
+
+					assert.equal(
+						log.length,
+						numChanges,
+						"Mutation after unsubscribe must not emit change events.",
+					);
+				},
+			);
 		}
 
 		describe("object", () => {
-			check(
-				makeSchema((_) =>
-					_.object("", {
-						content: _.boolean,
-					}),
-				),
-				{ content: false },
-				($) => ($.content = !$.content),
-			);
+			check(($) => $.object.content++);
```

Comment:
Whoop.  I meant to rename '$' -> 'root', as I think it's unnecessarily opaque.

(I am thinking we might want to standardize on '_' for SchemaBuilder in docs/examples, however, as it makes our 'schema.ts' files look almost as nice as packages that can use free functions.)

Reply:
I gotcha, renamed to root :)